adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java

```diff
@@ -109,6 +115,94 @@ public void testStoppedClusterIsNotAvailable() throws Exception {
     adapter.isAvailable();
   }
 
+  @Test
+  public void testResourceFileIsRead() throws URISyntaxException {
+    String testId = "order-66-clones-memo";
+    Date testDate = new Date();
+    String testName = String.format("%s_%s", testId, testDate.getTime());
+    String url = String.format("%s/webhdfs/v1/%s.txt", BASE_URL, testName);
+
+    // creates the resource in the HDFS instance to be read and verifies the resource exists
+    CreateStorageRequest createStorageRequest =
+        generateTestStorageRequest(testId, testName, testDate, testDate, new URI(url));
+    adapter.createResource(createStorageRequest);
+    verifyFileExists(testName + ".txt");
+
+    // gets the resource metadata to create a read request
+    Metadata resourceMetadata = createStorageRequest.getResources().get(0).getMetadata();
+    ResourceRequest readRequest = new ResourceRequestImpl(resourceMetadata);
+    ResourceResponse readResponse = adapter.readResource(readRequest);
+    Resource readResource = readResponse.getResource();
+
+    assertThat(readResource.getId(), is(testId));
+    assertThat(readResource.getName(), is(String.format("%s_%s", testId, testDate.getTime())));
+    // TODO - uncomment below line when file content issue is fixed
+    // assertThat(readInputStreamToString(readResource.getInputStream()), is("my-data"));
```

Comment:
My PR has a method called `verifyFileContents(String filename, String expected)` that will check the file with the given name against the expected string value. It could be useful here. Just depends on whether or not you want to keep it in.